rfd/0159-webasset-cache.md

```diff
@@ -0,0 +1,164 @@
+
+---
+authors: Michael Myers (michael.myers@goteleport.com)
+state: draft
+---
+
+# RFD 0159 - Webasset Cache
+
+## Required approvers
+
+
+## What
+
+We discuss the creation of an in-memory cache of webassets in the auth server to
+provide a way for proxies to serve files from different versions of Teleport.
+Here is a quick PoC https://www.loom.com/share/9983cae0e8574149a92dcec623e07f10
+
+## Why
+
+Up until now we've always thought of the 1-to-1 relationship of webasset bundle
+to proxy version was a given. In most cases, it still is. However, some
+customers may run multiple versions of the proxy behind a load balancer and this
```

Comment:
Could you explain why we are interested in supporting this use case?

For Cloud, we run multiple proxy instances with different versions at the same time during upgrades, but new web UI connections are not usually load balanced between those instances. (The version could obviously still change between requests.)